lib/push.js

```diff
@@ -35,25 +44,31 @@ module.exports = function (log, db) {
             if (device.pushCallback) {
               // send the push notification
               log.increment(LOG_UPDATE_INCREMENT_SEND)
-              request.post(device.pushCallback, function (err){
+              request.post(device.pushCallback, function (err, response) {
                 if (err) {
-                  log.error({
-                    op: LOG_OP_NOTIFY_UPDATE,
-                    deviceId: deviceId,
-                    err: err
-                  })
-                  log.increment(LOG_UPDATE_INCREMENT_FAIL)
+                  // any 400 level error from the push servers means
+                  // the push settings need to be reset.
+                  // the clients will check this and re-register push endpoints
```

Comment:
@kitcambridge  Hm if it is a 404 the client needs to fix it. i.e get a new one. Otherwise we will never able to reestablish device push for that device, no?